From 1e0d948344543e8536972cff8a70f2880d65917b Mon Sep 17 00:00:00 2001
From: Sunil Kumar Kori <skori@marvell.com>
Date: Fri, 5 Jan 2024 20:30:43 +0530
Subject: [PATCH 126/513] doc: update command scope information

Set of CLI commands are classified into following types;

- Commands which must be used in script only.
- Commands which must be used via telnet session only.
- Commands which can be used either in script or via telnet session.

Rename "Dynamic" column to "Scope" to provide clear scope of commands.

Signed-off-by: Sunil Kumar Kori <skori@marvell.com>
Change-Id: Id5965633303cbd1a0d0a9fd3da20f1b1bcfb733f
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/dataplane/dpdk/+/119244
Base-Builds: sa_ip-toolkits-Jenkins <sa_ip-toolkits-jenkins@marvell.com>
Base-Tests: sa_ip-toolkits-Jenkins <sa_ip-toolkits-jenkins@marvell.com>
Tested-by: sa_ip-toolkits-Jenkins <sa_ip-toolkits-jenkins@marvell.com>
Reviewed-by: Jerin Jacob Kollanukkaran <jerinj@marvell.com>
---
 doc/guides/tools/graph.rst | 211 +++++++++++++++++++------------------
 1 file changed, 108 insertions(+), 103 deletions(-)

diff --git a/doc/guides/tools/graph.rst b/doc/guides/tools/graph.rst
index 1855d1289183a..6c559afe3538b 100644
--- a/doc/guides/tools/graph.rst
+++ b/doc/guides/tools/graph.rst
@@ -154,109 +154,114 @@ file to express the requested use case configuration.
 .. table:: Exposed CLIs
    :widths: auto
 
-   +--------------------------------------+-----------------------------------+---------+----------+
-   |               Command                |             Description           | Dynamic | Optional |
-   +======================================+===================================+=========+==========+
-   | | graph <usecases> [bsz <size>]      | | Command to express the desired  |   No    |    No    |
-   | | [tmo <ns>] [coremask <bitmask>]    | | use case. Also enables/disable  |         |          |
-   | | model <rtc/mcd/default> pcap_enable| | pcap capturing.                 |         |          |
-   | | <0/1> num_pcap_pkts <num> pcap_file|                                   |         |          |
-   | | <output_capture_file>              |                                   |         |          |
-   +--------------------------------------+-----------------------------------+---------+----------+
-   | graph start                          | | Command to start the graph.     |   No    |    No    |
-   |                                      | | This command triggers that no   |         |          |
-   |                                      | | more commands are left to be    |         |          |
-   |                                      | | parsed and graph initialization |         |          |
-   |                                      | | can be started now. It must be  |         |          |
-   |                                      | | the last command in usecase.cli |         |          |
-   +--------------------------------------+-----------------------------------+---------+----------+
-   | graph stats show                     | | Command to dump current graph   |   Yes   |    Yes   |
-   |                                      | | statistics.                     |         |          |
-   +--------------------------------------+-----------------------------------+---------+----------+
-   | help graph                           | | Command to dump graph help      |   Yes   |    Yes   |
-   |                                      | | message.                        |         |          |
-   +--------------------------------------+-----------------------------------+---------+----------+
-   | | mempool <mempool_name> size        | | Command to create mempool which |   No    |    No    |
-   | | <mbuf_size> buffers                | | will be further associated to   |         |          |
-   | | <number_of_buffers>                | | RxQ to dequeue the packets.     |         |          |
-   | | cache <cache_size> numa <numa_id>  |                                   |         |          |
-   +--------------------------------------+-----------------------------------+---------+----------+
-   | help mempool                         | | Command to dump mempool help    |   Yes   |    Yes   |
-   |                                      | | message.                        |         |          |
-   +--------------------------------------+-----------------------------------+---------+----------+
-   | | ethdev <ethdev_name> rxq <n_queues>| | Command to create DPDK port with|   No    |    No    |
-   | | txq <n_queues> <mempool_name>      | | given number of Rx and Tx queues|         |          |
-   |                                      | | . Also attach RxQ with given    |         |          |
-   |                                      | | mempool. Each port can have     |         |          |
-   |                                      | | single mempool only i.e. all    |         |          |
-   |                                      | | RxQs will share the same mempool|         |          |
-   |                                      | | .                               |         |          |
-   +--------------------------------------+-----------------------------------+---------+----------+
-   | ethdev <ethdev_name> mtu <mtu_sz>    | | Command to configure MTU of DPDK|   Yes   |    Yes   |
-   |                                      | | port.                           |         |          |
-   +--------------------------------------+-----------------------------------+---------+----------+
-   |  | ethdev <ethdev_name> promiscuous  | | Command to enable/disable       |   Yes   |    Yes   |
-   |  | <on/off>                          | | promiscuous mode on DPDK port.  |         |          |
-   +--------------------------------------+-----------------------------------+---------+----------+
-   | ethdev <ethdev_name> show            | | Command to dump current ethdev  |   Yes   |    Yes   |
-   |                                      | | configuration.                  |         |          |
-   +--------------------------------------+-----------------------------------+---------+----------+
-   | ethdev <ethdev_name> stats           | | Command to dump current ethdev  |   Yes   |    Yes   |
-   |                                      | | statistics.                     |         |          |
-   +--------------------------------------+-----------------------------------+---------+----------+
-   | | ethdev <ethdev_name> ip4 addr add  | | Command to configure IPv4       |   Yes   |    Yes   |
-   | | <ip> netmask <mask>                | | address on given PCI device. It |         |          |
-   |                                      | | is needed if user wishes to use |         |          |
-   |                                      | | ``ipv4_lookup`` node.           |         |          |
-   +--------------------------------------+-----------------------------------+---------+----------+
-   | | ethdev <ethdev_name> ip6 addr add  | | Command to configure IPv6       |   Yes   |    Yes   |
-   | | <ip> netmask <mask>                | | address on given PCI device. It |         |          |
-   |                                      | | is needed if user wishes to use |         |          |
-   |                                      | | ``ipv6_lookup`` node.           |         |          |
-   +--------------------------------------+-----------------------------------+---------+----------+
-   | help ethdev                          | | Command to dump ethdev help     |   Yes   |    Yes   |
-   |                                      | | message.                        |         |          |
-   +--------------------------------------+-----------------------------------+---------+----------+
-   | | ipv4_lookup route add ipv4 <ip>    | | Command to add a route into     |   Yes   |    Yes   |
-   | |  netmask <mask> via <ip>           | | ``ipv4_lookup`` LPM table. It is|         |          |
-   |                                      | | needed if user wishes to route  |         |          |
-   |                                      | | the packets based on LPM lookup |         |          |
-   |                                      | | table.                          |         |          |
-   +--------------------------------------+-----------------------------------+---------+----------+
-   | help ipv4_lookup                     | | Command to dump ``ipv4_lookup`` |   Yes   |    Yes   |
-   |                                      | | help message.                   |         |          |
-   +--------------------------------------+-----------------------------------+---------+----------+
-   | | ipv6_lookup route add ipv6 <ip>    | | Command to add a route into     |   Yes   |    Yes   |
-   | |  netmask <mask> via <ip>           | | ``ipv6_lookup`` LPM table. It is|         |          |
-   |                                      | | needed if user wishes to route  |         |          |
-   |                                      | | the packets based on LPM6 lookup|         |          |
-   |                                      | | table.                          |         |          |
-   +--------------------------------------+-----------------------------------+---------+----------+
-   | help ipv6_lookup                     | | Command to dump ``ipv6_lookup`` |   Yes   |    Yes   |
-   |                                      | | help message.                   |         |          |
-   +--------------------------------------+-----------------------------------+---------+----------+
-   | neigh add ipv4 <ip> <mac>            | | Command to add a neighbour      |   Yes   |    Yes   |
-   |                                      | | information into                |         |          |
-   |                                      | | ``ipv4_rewrite`` node.          |         |          |
-   +--------------------------------------+-----------------------------------+---------+----------+
-   | neigh add ipv6 <ip> <mac>            | | Command to add a neighbour      |   Yes   |    Yes   |
-   |                                      | | information into                |         |          |
-   |                                      | | ``ipv6_rewrite`` node.          |         |          |
-   +--------------------------------------+-----------------------------------+---------+----------+
-   | help neigh                           | | Command to dump neigh help      |   Yes   |    Yes   |
-   |                                      | | message.                        |         |          |
-   +--------------------------------------+-----------------------------------+---------+----------+
-   | | ethdev_rx map port <ethdev_name>   | | Command to add port-queue-core  |   No    |    No    |
-   | | queue <q_num> core <core_id>       | | mapping to ``ethdev_rx`` node.  |         |          |
-   |                                      | | ``ethdev_rx`` node instance will|         |          |
-   |                                      | | be pinned on given core and will|         |          |
-   |                                      | | poll on requested port/queue    |         |          |
-   |                                      | | pair.                           |         |          |
-   +--------------------------------------+-----------------------------------+---------+----------+
-   | help ethdev_rx                       | | Command to dump ethdev_rx help  |   Yes   |    Yes   |
-   |                                      | | message.                        |         |          |
-   +--------------------------------------+-----------------------------------+---------+----------+
-
+   +--------------------------------------+-----------------------------------+-------------------+----------+
+   |               Command                |             Description           |     Scope         | Optional |
+   +======================================+===================================+===================+==========+
+   | | graph <usecases> [bsz <size>]      | | Command to express the desired  | :ref:`1 <scopes>` |    No    |
+   | | [tmo <ns>] [coremask <bitmask>]    | | use case. Also enables/disable  |                   |          |
+   | | model <rtc/mcd/default> pcap_enable| | pcap capturing.                 |                   |          |
+   | | <0/1> num_pcap_pkts <num> pcap_file|                                   |                   |          |
+   | | <output_capture_file>              |                                   |                   |          |
+   +--------------------------------------+-----------------------------------+-------------------+----------+
+   | graph start                          | | Command to start the graph.     | :ref:`1 <scopes>` |    No    |
+   |                                      | | This command triggers that no   |                   |          |
+   |                                      | | more commands are left to be    |                   |          |
+   |                                      | | parsed and graph initialization |                   |          |
+   |                                      | | can be started now. It must be  |                   |          |
+   |                                      | | the last command in usecase.cli |                   |          |
+   +--------------------------------------+-----------------------------------+-------------------+----------+
+   | graph stats show                     | | Command to dump current graph   | :ref:`2 <scopes>` |    Yes   |
+   |                                      | | statistics.                     |                   |          |
+   +--------------------------------------+-----------------------------------+-------------------+----------+
+   | help graph                           | | Command to dump graph help      | :ref:`2 <scopes>` |    Yes   |
+   |                                      | | message.                        |                   |          |
+   +--------------------------------------+-----------------------------------+-------------------+----------+
+   | | mempool <mempool_name> size        | | Command to create mempool which | :ref:`1 <scopes>` |    No    |
+   | | <mbuf_size> buffers                | | will be further associated to   |                   |          |
+   | | <number_of_buffers>                | | RxQ to dequeue the packets.     |                   |          |
+   | | cache <cache_size> numa <numa_id>  |                                   |                   |          |
+   +--------------------------------------+-----------------------------------+-------------------+----------+
+   | help mempool                         | | Command to dump mempool help    | :ref:`2 <scopes>` |    Yes   |
+   |                                      | | message.                        |                   |          |
+   +--------------------------------------+-----------------------------------+-------------------+----------+
+   | | ethdev <ethdev_name> rxq <n_queues>| | Command to create DPDK port with| :ref:`1 <scopes>` |    No    |
+   | | txq <n_queues> <mempool_name>      | | given number of Rx and Tx queues|                   |          |
+   |                                      | | . Also attach RxQ with given    |                   |          |
+   |                                      | | mempool. Each port can have     |                   |          |
+   |                                      | | single mempool only i.e. all    |                   |          |
+   |                                      | | RxQs will share the same mempool|                   |          |
+   |                                      | | .                               |                   |          |
+   +--------------------------------------+-----------------------------------+-------------------+----------+
+   | ethdev <ethdev_name> mtu <mtu_sz>    | | Command to configure MTU of DPDK| :ref:`3 <scopes>` |    Yes   |
+   |                                      | | port.                           |                   |          |
+   +--------------------------------------+-----------------------------------+-------------------+----------+
+   | | ethdev <ethdev_name> promiscuous   | | Command to enable/disable       | :ref:`3 <scopes>` |    Yes   |
+   | | <on/off>                           | | promiscuous mode on DPDK port.  |                   |          |
+   +--------------------------------------+-----------------------------------+-------------------+----------+
+   | ethdev <ethdev_name> show            | | Command to dump current ethdev  | :ref:`2 <scopes>` |    Yes   |
+   |                                      | | configuration.                  |                   |          |
+   +--------------------------------------+-----------------------------------+-------------------+----------+
+   | ethdev <ethdev_name> stats           | | Command to dump current ethdev  | :ref:`2 <scopes>` |    Yes   |
+   |                                      | | statistics.                     |                   |          |
+   +--------------------------------------+-----------------------------------+-------------------+----------+
+   | | ethdev <ethdev_name> ip4 addr add  | | Command to configure IPv4       | :ref:`3 <scopes>` |    Yes   |
+   | | <ip> netmask <mask>                | | address on given PCI device. It |                   |          |
+   |                                      | | is needed if user wishes to use |                   |          |
+   |                                      | | ``ipv4_lookup`` node.           |                   |          |
+   +--------------------------------------+-----------------------------------+-------------------+----------+
+   | | ethdev <ethdev_name> ip6 addr add  | | Command to configure IPv6       | :ref:`3 <scopes>` |    Yes   |
+   | | <ip> netmask <mask>                | | address on given PCI device. It |                   |          |
+   |                                      | | is needed if user wishes to use |                   |          |
+   |                                      | | ``ipv6_lookup`` node.           |                   |          |
+   +--------------------------------------+-----------------------------------+-------------------+----------+
+   | help ethdev                          | | Command to dump ethdev help     | :ref:`2 <scopes>` |    Yes   |
+   |                                      | | message.                        |                   |          |
+   +--------------------------------------+-----------------------------------+-------------------+----------+
+   | | ipv4_lookup route add ipv4 <ip>    | | Command to add a route into     | :ref:`3 <scopes>` |    Yes   |
+   | |  netmask <mask> via <ip>           | | ``ipv4_lookup`` LPM table. It is|                   |          |
+   |                                      | | needed if user wishes to route  |                   |          |
+   |                                      | | the packets based on LPM lookup |                   |          |
+   |                                      | | table.                          |                   |          |
+   +--------------------------------------+-----------------------------------+-------------------+----------+
+   | help ipv4_lookup                     | | Command to dump ``ipv4_lookup`` | :ref:`2 <scopes>` |    Yes   |
+   |                                      | | help message.                   |                   |          |
+   +--------------------------------------+-----------------------------------+-------------------+----------+
+   | | ipv6_lookup route add ipv6 <ip>    | | Command to add a route into     | :ref:`3 <scopes>` |    Yes   |
+   | |  netmask <mask> via <ip>           | | ``ipv6_lookup`` LPM table. It is|                   |          |
+   |                                      | | needed if user wishes to route  |                   |          |
+   |                                      | | the packets based on LPM6 lookup|                   |          |
+   |                                      | | table.                          |                   |          |
+   +--------------------------------------+-----------------------------------+-------------------+----------+
+   | help ipv6_lookup                     | | Command to dump ``ipv6_lookup`` | :ref:`2 <scopes>` |    Yes   |
+   |                                      | | help message.                   |                   |          |
+   +--------------------------------------+-----------------------------------+-------------------+----------+
+   | neigh add ipv4 <ip> <mac>            | | Command to add a neighbour      | :ref:`3 <scopes>` |    Yes   |
+   |                                      | | information into                |                   |          |
+   |                                      | | ``ipv4_rewrite`` node.          |                   |          |
+   +--------------------------------------+-----------------------------------+-------------------+----------+
+   | neigh add ipv6 <ip> <mac>            | | Command to add a neighbour      | :ref:`3 <scopes>` |    Yes   |
+   |                                      | | information into                |                   |          |
+   |                                      | | ``ipv6_rewrite`` node.          |                   |          |
+   +--------------------------------------+-----------------------------------+-------------------+----------+
+   | help neigh                           | | Command to dump neigh help      | :ref:`2 <scopes>` |    Yes   |
+   |                                      | | message.                        |                   |          |
+   +--------------------------------------+-----------------------------------+-------------------+----------+
+   | | ethdev_rx map port <ethdev_name>   | | Command to add port-queue-core  | :ref:`1 <scopes>` |    No    |
+   | | queue <q_num> core <core_id>       | | mapping to ``ethdev_rx`` node.  |                   |          |
+   |                                      | | ``ethdev_rx`` node instance will|                   |          |
+   |                                      | | be pinned on given core and will|                   |          |
+   |                                      | | poll on requested port/queue    |                   |          |
+   |                                      | | pair.                           |                   |          |
+   +--------------------------------------+-----------------------------------+-------------------+----------+
+   | help ethdev_rx                       | | Command to dump ethdev_rx help  | :ref:`2 <scopes>` |    Yes   |
+   |                                      | | message.                        |                   |          |
+   +--------------------------------------+-----------------------------------+-------------------+----------+
+
+.. _scopes:
+
+1. Script only
+2. Telnet only
+3. Script and telnet both
 
 Runtime configuration
 ---------------------
-- 
2.25.1

